xapian-letor/api/featurelist_internal.cc

```diff
@@ -162,33 +164,30 @@ FeatureList::Internal::compute_collection_termfreq()
 }
 
 void
-FeatureList::Internal::populate_feature(Feature *feature_)
+FeatureList::Internal::populate_featurelist(Feature::Internal *internal_feature)
 {
-    stat_flags stats_needed = stat_flags(feature_->get_stats());
+    for (Feature* it : feature) {
+	stats_needed = stat_flags(stats_needed | it->get_stats());
+    }
```

Comment:
I'm pretty sure this can (and should) be done in the constructor, called when the `FeatureList` itself is constructed (which is when the list of `Feature` objects is put together / copied in). As it is, doing it here suggests that it can change every time we populate, which isn't true. (Indeed, it can't be true because you aren't clearing `stats_needed`.)

Reply:
You are right. I missed it. fixed in a fixup commit.